packages/api/src/api/auth/gitHub.ts

```diff
@@ -0,0 +1,93 @@
+import { Router } from 'express';
+import { env } from '../../env';
+
+const passport = require('passport');
+const GitHubStrategy = require('passport-github2').Strategy;
+const session = require('express-session'); // express-session
+
+const GitHubClientId = env.GitHubId;
+const GitHubClientSecret = env.GitHubSecret;
+
+export const github = Router();
+
+passport.serializeUser((user: any, done: any) => {
+  done(null, user.id);
+});
+
+passport.deserializeUser((id: any, done: any) => {
+  done(null, id);
+});
+
+github.use(
+  // express-session
+  session({
+    secret: 'keyboard cat',
+    resave: false,
+    saveUninitialized: false,
+    cookie: {
+      httpOnly: false,
+      secure: false,
+      maxAge: 24 * 60 * 60 * 1000, // 1 day long
+    },
+  }),
+);
+
+github.use(passport.initialize());
+github.use(passport.session());
+
+passport.use(
+  new GitHubStrategy(
+    {
+      clientID: GitHubClientId,
+      clientSecret: GitHubClientSecret,
+      callbackURL: 'http://localhost:3000/api/auth/github/callback',
+    },
+    async (accessToken: any, refreshToken: any, profile: any, done: any) => {
+      const user = {
+        name: profile.username,
+        gitHubId: profile.id,
+      };
+
+      const res = await fetch(`http://localhost:3000/api/users/${profile.id}`);
+      if (res.status === 404) {
+        // if we dont find the user, we do this block
+        const res2 = await fetch('http://localhost:3000/api/users', {
+          method: 'POST',
+          headers: {
+            'Content-Type': 'application/json',
+          },
+          body: JSON.stringify(user),
+        });
+
+        if (res2.status === 201) {
+          // user created, now we go to this block
+          done(null, profile);
+        }
+      } else {
+        // if user is already existing, we execute this
+        done(null, profile);
+      }
```

Comment:
You're on the server, you don't need to request from within the server you just need to query the database.